droidmediacodec.cpp

```diff
@@ -317,17 +335,95 @@ class DroidMediaCodecBuilder {
     if (md == NULL) {
       return NULL;
     }
-#if ANDROID_MAJOR < 7
-    return android::OMXCodec::Create(omx.interface(),
-				     md,
-				     isEncoder(),
-				     src,
-				     NULL, flags(), window);
+
+#if ANDROID_MAJOR >= 5
+    const char *mime;
+    md->findCString(android::kKeyMIMEType, &mime);
+
+    if (isEncoder()) {
+      if (!strncmp("video", mime, 5)) {
+      android::sp<android::AMessage> format = new android::AMessage();
+
+      format->setString("mime", mime);
+      ALOGW("Creating video encoder for %s", mime);
+      int32_t width, height, stride, sliceh, colour, frames, maxinput, bitrate, iframes;
+
+      md->findInt32(android::kKeyWidth, &width);
```

Comment:
I wonder if we should have if (md->findInt32()) format->setInt32()? Do we need to check if width > 0 here?

Reply:
Yes I think that would be simpler. Will change.